src/asm/parser.y

```diff
@@ -1861,7 +1860,14 @@ z80_ldio	: T_Z80_LDH T_MODE_A T_COMMA op_mem_ind {
 ;
 
 c_ind		: T_LBRACK T_MODE_C T_RBRACK
-		| T_LBRACK T_MODE_HW_C T_RBRACK
+		| T_LBRACK relocexpr T_OP_ADD T_MODE_C T_RBRACK {
+			int32_t val = rpn_GetConstVal(&$2);
+
+			if (val != 0xff00) {
```

Comment:
To avoid potentially two error messages here, I would do:
```
if (!rpn_isKnown(&$2) || $2.val != 0xff00)
    error("Expected constant expression equal to $FF00 for \"$ff00+c\"");
```

Reply:
(The "`(did you mistype the C?)`" is also unusual for RGBDS error messages.)

Reply:
Unusual, certainly, but is that a bad thing? This is a weird edge case of the parser, so guiding users into likely error causes should be a win, right?

Reply:
I think the "equal to $FF00" is clear enough if they try something like `ld [$c200+c], a`.

Reply:
I'm considering
```
cd = 420

ld [$C200 + c], a ; Oops, I forgot the `d` in `cd`!
```

Reply:
Then they look at the line and see the `+c` typo. `bg equ $42 / ld [$c200+b], a` just says "syntax error, unexpected `b`".

Reply:
Right, I guess.

Reply:
I wouldn't really mind if "did you mistype the C?" were very likely to be helpful advice, but it's misleading in the cases where the not-$ff00-number is the mistake.